File: internal/dto/scitech2025.go
package dto

import (
	"time"

	"github.com/google/uuid"
)

type SciTech2025RegistrationRequest struct {
	FullName            string  `json:"full_name"`
	Email               string  `json:"email"`
	Country             string  `json:"country"`
	Organization        string  `json:"organization"`
	ParticipationFormat string  `json:"participation_format"`
	FullNameRU          *string `json:"full_name_ru"`
	IsPresenter         bool    `json:"is_presenter"`
	ConfChairPaperID    *string `json:"confchair_paper_id"`
	PaperTitle          *string `json:"paper_title"`
}

type SciTech2025RegistrationResponse struct {
	ID                  uuid.UUID `json:"id"`
	FullName            string    `json:"full_name"`
	Email               string    `json:"email"`
	Country             string    `json:"country"`
	Organization        string    `json:"organization"`
	ParticipationFormat string    `json:"participation_format"`
	FullNameRU          *string   `json:"full_name_ru,omitempty"`
	IsPresenter         bool      `json:"is_presenter"`
	ConfChairPaperID    *string   `json:"confchair_paper_id,omitempty"`
	PaperTitle          *string   `json:"paper_title,omitempty"`
	CreatedAt           time.Time `json:"created_at"`
}

File: src/api/services/scitech2025.ts
import axios from "axios";
import type { BaseAPIResponse } from "../types";

const apiInstance = axios.create({
	baseURL: `${import.meta.env.VITE_API_URL}/scitech2025`,
	timeout: 1000,
	headers: {
		"Content-Type": "application/json",
	},
});
export interface RegistrationRequest {
	full_name: string;
	email: string;
	country: string;
	organization: string;
	participation_format: string;
	is_presenter: boolean;
	full_name_ru?: string | undefined;
	confchair_paper_id?: string | undefined;
	paper_title?: string | undefined;
}
export interface RegistrationResponse extends RegistrationRequest {
	id: string;
	updated_at: string;
}
export const SciTech2025Service = {
	async register(data: RegistrationRequest) {
		const res = await apiInstance.post<
			BaseAPIResponse<RegistrationResponse>
		>(`/register`, data);
		return res.data.data;
	},
};
